scripts/EGAP/create_EGAP_json.py

```diff
@@ -0,0 +1,253 @@
+import logging
+import csv
+import datetime
+import json
+import os
+import shutil
+import re
+import jsonschema
+
+from django.core.management.base import BaseCommand
+from jsonschema.exceptions import ValidationError
+
+from website.project.metadata.utils import create_jsonschema_from_metaschema
+from website.project.metadata.schemas import ensure_schema_structure, from_json
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+schema_to_spreadsheet_mapping = [
+    {'q1': 'TITLE'},
+    {'q2': 'B2 AUTHORS'},
+    {'q3': 'ID'},
+    {'q4': 'POST DATE'},
+    {'q5': 'B3 ACKNOWLEDGEMENTS'},
+    {'q6': 'B4 FACULTY MEMBER?'},
+    {'q8': 'B5 PROSPECTIVE OR RETROSPECTIVE?'},
+    {'q10': 'B6 EXPERIMENTAL STUDY?'},
+    {'q11': 'B7 DATE OF START OF STUDY'},
+    {'q12': 'B8 GATE DATE'},
+    {'q13': 'B9 PRESENTED AT EGAP MEETING?'},
+    {'q14': 'B10 PRE-ANALYSIS PLAN WITH REGISTRATION?'},
+    {'q15': 'C1 BACKGROUND'},
+    {'q16': 'C2 HYPOTHESES'},
+    {'q17': 'C3 TESTING PLAN'},
+    {'q18': 'C4 COUNTRY'},
+    {'q19': 'C5 SAMPLE SIZE'},
+    {'q20': 'C6 POWER ANALYSIS?'},
+    {'q22': 'C7 IRB APPROVAL?'},
+    {'q24': 'C8 IRB NUMBER'},
+    {'q25': 'C9 DATE OF IRB APPROVAL'},
+    {'q26': 'C10 INTERVENTION IMPLEMENTER'},
+    {'q28': 'C11 REMUNERATION?'},
+    {'q30': 'C12 PUBLICATION AGREEMENT?'},
+    {'q32': 'C13 JEL CODES'},
+    {'q33': 'METHODOLOGY'},
+    {'q34': 'POLICY'},
+]
+
+# Any multiple choice questions where "Other" is a possible response, have subsequent "Other"
+# question to log that response.  If multiple choice question value is invalid,
+# attempt to log the value in the corresponding "Other" question response.
+other_mapping = {
+    'q6': 'q7',
+    'q8': 'q9',
+    'q20': 'q21',
+    'q22': 'q23',
+    'q26': 'q27',
+    'q28': 'q29',
+    'q30': 'q31'
+}
+
+def create_file_tree_and_json():
+    # Things this function needs to do:
+    # For each row in the registry function, create a directory.
+    # Create two JSON files, one project json with ID, Title, Postdate, and authors listed
+    # with emails. And another with all the key value pairs for the registry meta.
+    top_dir = 'EGAP_data_{}'.format(datetime.datetime.now().strftime('%m-%d-%Y'))
+    logger.info('Creating EGAP directory at {}'.format(top_dir))
+    os.mkdir(top_dir)
+    author_list = create_author_dict()
+    with open('EGAP_registry_for_OSF.csv') as csv_registry_file:
+        csv_reader = csv.reader(csv_registry_file, delimiter=',')
+        header_row = next(csv_reader)
+        normalized_header_row = [col_header.decode('ascii', 'ignore') for col_header in header_row]
+
+        id_index = normalized_header_row.index('ID')
+        for line in csv_reader:
+            row = [cell.decode('ascii', 'ignore').strip() for cell in line]
```

Comment:
No UTF-8 in this spreadsheet? I wonder if that's guaranteed. Consider [utf-8 decoding in the csv parsing](https://stackoverflow.com/a/904085/3579517) (I don't know if it's worth it for this, mind).

Reply:
I tried to change this to unicode(cell, UTF-8) but encountered errors.